core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -457,7 +457,7 @@ public void purgeExpired() {
             checkStoreAvailability();
             Consumer<CacheWriter> purgeWriter = writer -> {
                // ISPN-6711 Shared stores should only be purged by the coordinator
-               if (getStoreConfig(writer).shared() && !transport.isCoordinator())
+               if (getStoreConfig(writer).shared() && transport != null && !transport.isCoordinator())
```

Comment:
instead of ```transport!=null```, could we use ```globalConfiguration.isClustered()``` ? when reading the line, it seems like we are avoiding a null pointer exception (like protecting) for an eventual transport null instead of something that explains that indeed it needs to be clustered to check it's a coordinator

Reply:
Good point, that is much clearer